Mobile/ShopLens/Droid/CameraActivity.cs

```diff
@@ -0,0 +1,101 @@
+﻿
+using Uri = Android.Net.Uri;
+using System;
+using System.Collections.Generic;
+using Java.IO;
+
+using Android.Provider;
+using Android.Graphics;
+using Android.Content.PM;
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Widget;
+using Camera;
+using Android.Support.V4.Content;
+using Android.Support.Compat;
+
+namespace ShopLens.Droid
+{
+
+    [Activity (Label = "CameraActivity")]
+    public class CameraActivity : Activity
+    {
+        Button BtnTakeImg;
+        ImageView ImgView;
+
+        public static File productPhoto;
+        public static File _dir;
+
+        public static int REQUEST_IMAGE = 102;
+        public static string FILE_PROVIDER_NAME = ".shoplens.fileprovider";
+
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+
+            base.OnCreate(savedInstanceState);
+
+            SetContentView(Resource.Layout.Camera);
+
+            if (IsThereAnAppToTakePictures())
+            {
+                CreateDirectoryForPictures();
+                BtnTakeImg = FindViewById<Button>(Resource.Id.btntakepicture);
+                ImgView = FindViewById<ImageView>(Resource.Id.ImgTakeimg);
+                BtnTakeImg.Click += TakeAPicture;
+            }
+        }
+
+        private void CreateDirectoryForPictures()
```

Comment:
Maybe we should have a seperate class for I.O. operations, but this doesn't seem too out-of-place here either